bin/prover-client/src/args.rs

```diff
@@ -66,4 +106,22 @@ impl Args {
     pub fn get_btc_rpc_url(&self) -> String {
         format!("http://{}", self.bitcoind_url)
     }
+
+    pub fn get_workers(&self) -> HashMap<ProofZkVm, usize> {
+        let mut workers = HashMap::new();
+
+        workers.insert(ProofZkVm::Native, self.native_workers);
+
+        #[cfg(feature = "sp1")]
+        {
+            workers.insert(ProofZkVm::SP1, self.sp1_workers);
+        }
+
+        #[cfg(feature = "risc0")]
+        {
+            workers.insert(ProofZkVm::Risc0, self.risc0_workers);
+        }
+
+        workers
+    }
```

Comment:
I know we discussed this. I gave some thought. I still don't like implementation details to be hardcoded in abstractions like traits.

This should be moved into the `impl`s themselves:

```suggestion

    pub fn get_workers(&self) -> HashMap<ProofZkVm, usize>;
```

Reply:
Also docstrings. VERY IMPORTANT in traits.

Reply:
I'm a bit confused. This is not a trait.